databricks-sdk-java/src/main/java/com/databricks/sdk/core/NotebookNativeCredentialsProvider.java

```diff
@@ -0,0 +1,133 @@
+package com.databricks.sdk.core;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A CredentialsProvider that uses the API token from the command context to authenticate.
+ *
+ * <p>The token and hostname are read from the command context, which can be retrieved through the
+ * dbutils API. As the Java SDK does not depend on DBUtils directly, reflection is used to retrieve
+ * the token. This token should be available wherever the DBUtils API is accessible (i.e. in the
+ * Spark driver).
+ */
+public class NotebookNativeCredentialsProvider implements CredentialsProvider {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(NotebookNativeCredentialsProvider.class);
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  @Override
+  public String authType() {
+    return "runtime";
```

Comment:
Thought -- I think `runtime` auth here might be confusing, do we want to have `notebook-native` auth type and alias that to `runtime`? But if we decide to go in this direction then this would need to be backported to python SDK as well and there probably isn't high ROI to do this.

Reply:
This is already implemented in the Python SDK, and the auth type there is called `runtime`: https://github.com/databricks/databricks-sdk-py/blob/main/databricks/sdk/core.py#L95